### PR TITLE
Dynamic jwt_cert_key for unit tests

### PR DESCRIPTION
**Pull request by abottoms-coder**
### Summary of Changes
This ticket is to dynamically generate the environment variable that our unit tests use for accessing the API so as to prevent us needing to hardcode a value in the .env file. 
_Addresses issue #969

Once merged, we can and should delete the JWT_CERT_KEY env variables in both our cloud platforms CircleCI and Cloud.gov.

### How to Test
_List the steps to test the PR_
In a local development environment, open TANF-app/tdrs-backend/.env and delete the JWT_CERT_KEY line:
`JWT_CERT_KEY=...`

Now run the tests using this command:
`docker-compose run --rm web bash -c pytest`
This command will run all unit tests including the refactored ones without being supplied the environment variable JWT_CERT_KEY which was previously needed.

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [x] Check against the criteria:
+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
- [x]  JWT_KEY (JWT_CERT_TEST) used for tests is dynamically generated along with any needed access tokens
- [x]  Tests are updated to be more DRY
- [x]  Testing Checklist has been run and all tests pass
- [ ]  README is updated, if necessary

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [x] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/a11y/how-18f-will-test-a11y.md)

_n/a:backend changes only_
### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [x] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?
_no security issues detected_

### CODE REVIEW

File: tdrs-backend/tdpservice/users/test/test_auth.py [hunk 1]
```diff
@@ -153,486 +150,326 @@ def test_login_fails_without_state(api_client):
...
-    api_client,
-    mock_token,
-    mocker,
-    patch_login_gov_jwt_key,
-    settings,
-    user
+def decoded_token(
+    email,
+    nonce,
+    sub="b2d2d115-1d7e-4579-b9d6-f8e84f4f56ca",
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/test/test_auth.py:180]:
```suggestion
    sub="uuid.uuidv4()",
```
This *might* work too but I'm not 100% sure.

File: tdrs-backend/tdpservice/users/test/test_auth.py [hunk 2]
```diff
@@ -153,486 +150,326 @@ def test_login_fails_without_state(api_client):
...
+        """Login should work with existing user."""
+        user.username = "test_old_email@example.com"
+        user.save()
+        states = states_factory
+        request = req_factory
+        request = create_session(request, states_factory)
+        view = TokenAuthorizationOIDC.as_view()
+        mock_post, mock_decode = mock
+        mock_decode.return_value = decoded_token(
+                    "test_new_email@example.com",
```
Comment (jtwillis92) [inline tdrs-backend/tdpservice/users/test/test_auth.py:356]:
```suggestion
                   user.username,
```

**Review by jtwillis92 — APPROVED** [review]
LGTM, left a couple small nits but this is a huge improvement!

**Review by riatzukiza — APPROVED** [review]
LGTM

**Pull request merged**